### PR TITLE
fix(executor): payloads are logged twice in Combiner

### LINKED ISSUE #5116: requests and responses are logged double times in Combiner node
We recently upgraded Seldon-core from 1.14 to 1,16. However we found some issues with payload logging.

## Describe the bug

This is our graph structure:
```
    graph:
      children:
      - endpoint:
          type: REST
        name: b
        type: MODEL
      - endpoint:
          type: REST
        name: c
        type: MODEL
      - endpoint:
          type: REST
        name: d
        type: MODEL
      - endpoint:
          type: REST
        name: e
        type: MODEL
      endpoint:
        type: REST
      logger:
        mode: all
        url: <URL>
      name: a
      type: COMBINER
    name: default
```

We have a combiner with 4 children. we only need to log the combiner(entry)'s requests and responses. 
However, requests and responses are logged double times in Combiner.

## To reproduce

If the seldon-core >= 1.16 and there is a combiner in the graph with logger enabled, the payloads of the combiner will be logged twice.

## Expected behaviour

each request and response should be only logged once in Combiner.

### PR DESCRIPTION
**Pull request by caozhuozi**
**What this PR does / why we need it**:
The PR aims to fix Issue #5116.
The bug descriprtion is mirrored below for easy reference.
If the seldon-core >= 1.16 and there is a combiner in the graph with logger enabled, the payloads of the Combiner will be logged twice.

**Which issue(s) this PR fixes**:

Fixes #5116

**Special notes for your reviewer**:
As I mentioned on the issue page, its better to first clarify/define the concept of *operation/situation*. In other words, we need to figure out:

 **What is a  _opration/situation_ during a payload lifecycle within a graph node?** 

With a clear agreement on what the concept refers to, we can then move to the log rule/convetion proposed by @cliveseldon.
> the key is the ability to log any before and after situation so as long as that can happen while at the same time removing redundant calls

 Otherwise redundant logs are inevitable if we  log payloads wherever we consider an operation/situation. 
 
 From my point of view, we can clarify *situation/operation* by leveraging the logic of the `Predict` method as it outlines the complete life cycle of a payload within a node.
 See my comments below for details.
```go
func (p *PredictorProcess) Predict(node *v1.PredictiveUnit, msg payload.SeldonPayload) (payload.SeldonPayload, error) {
        // ...

        // payloads before and after transformation input operation are logged inside
	tmsg, err := p.transformInput(node, msg, puid)
	if err != nil {
		return tmsg, err
	}
        // Payloads before calling Children nodes and after collection/aggregation of children nodes' results are also logged
        // We don't need to log payloads again within aggregate
	cmsg, err := p.predictChildren(node, tmsg, puid)
	if err != nil {
		return cmsg, err
	}
        // payloads before and after transformation output operation are logged inside
	response, err := p.transformOutput(node, cmsg, puid)

        // ...
	return response, err
}
```
To summarize, as per the `Predict` method, a playload lifecycle within a graph node can be divided into 3 phases: `transformInput`, `predictChildren` and `transformOutput`.  **Each phase can be regardad as a _situation_**.
Note that if we restrict the definition of *siuation* to first function call level within `Predict` method, for the `predictChildren` method, no matter what type of node is `ROUTER` or `COMBINER`, we should just get the payloads log before sending payload to children nodes and after collecting/aggregating results from children nodes respectively.

Thus, we don't need to do redundant payload  logging  in the inner  `aggregate` or `route` function anymore.

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @caozhuozi. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
@caozhuozi: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 160212b525ff0f9fddaae84d2a0f1c27c6021ae1 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-5123-integration-f4xl7) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@caozhuozi: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 160212b525ff0f9fddaae84d2a0f1c27c6021ae1 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-5123-integration-cjrqj) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
